R/EDGE-Industry.R

```diff
@@ -0,0 +1,2518 @@
+#' EDGE-Industry
+#' 
+#' Functions for calculating industry activity trajectories.
+#' 
+#' @md
+#' @param subtype One of
+#'   - `production`  Returns trajectories of primary and secondary steel
+#'     production
+#'   - `secondary.steel.max.share`  Returns the maximum share of secondary steel
+#'     in total steel production.
+#' @param match.steel.historic.values Should steel production trajectories match 
+#'   historic values?
+#' @param match.steel.estimates Should steel production trajectories match 
+#'   exogenous estimates?  `NULL` or one of
+#'   - `IEA_ETP` IEA 2017 Energy Transition Pathways steel production totals for
+#'     OECD and Non-OECD countries from the _Reference Technologies Scenario_
+#'     until 2060, and original growth rates after that.
+#' @param save.plots Save plots to `getConfig('outputfolder')`?
+#' @param China_Production A data frame with columns `period` and 
+#'     `total.production` prescribing total production for China to have, 
+#'     disregarding results from the stock saturation model.
+#'
+#' @return A list with a [`magpie`][magclass::magclass] object `x`, `weight`,
+#'   `unit`, `description`, `min`, and `max`.
+#'   
+#' @author Michaja Pehl
+#'
+#' @seealso [`calcOutput()`]
+#'
+#' @importFrom assertr assert verify within_bounds
+#' @importFrom broom tidy
+#' @importFrom car logit
+#' @importFrom dplyr %>% case_when right_join semi_join
+#' @importFrom Hmisc wtd.quantile
+#' @importFrom ggplot2 aes expand_limits facet_wrap geom_area geom_line 
+#'   geom_path geom_point ggplot ggsave guide_legend labs scale_fill_discrete 
+#'   scale_linetype_manual scale_shape_manual theme theme_minimal 
+#' @importFrom madrat calcOutput readSource toolGetMapping
+#' @importFrom quitte calc_mode duplicate duplicate_ list_to_data_frame 
+#'   madrat_mule order.levels sum_total_
+#' @importFrom readr write_rds
+#' @importFrom stats nls SSlogis sd
+#' @importFrom tidyr pivot_longer pivot_wider
+#' @importFrom zoo na.approx rollmean
+
+#' @rdname EDGE-Industry
+#' @export
+calcSteel_Projections <- function(subtype = 'production',
```

Comment:
why don't you call this file calcSteel_Projections? This would ensure full functionality from madrat

Reply:
What has `madrat` functionality to do with the file name?

Reply:
e.g. check of functions for caching

Reply:
And how is that, pray tell?  How does the name of a file, from which a package is compiled, influence either content or functionality of said package?

Reply:
based on the name it is checked whether the information from the cache can be loaded or it needs to be updated (if not started with an empty cache